Sources/XcodesKit/Environment.swift

```diff
@@ -53,6 +53,75 @@ public struct Shell {
     public func xcodeSelectSwitch(password: String?, path: String) -> Promise<ProcessOutput> {
         xcodeSelectSwitch(password, path)
     }
+    
+    public var downloadWithAria2: (Path, URL, Path, [HTTPCookie]) -> (Progress, Promise<Void>) = { aria2Path, url, destination, cookies in
+        let process = Process()
+        process.executableURL = aria2Path.url
+        process.arguments = [
+            "--header=Cookie: \(cookies.map { "\($0.name)=\($0.value)" }.joined(separator: "; "))",     
+            "--max-connection-per-server=16",
+            "--split=16",
+            "--summary-interval=1",
+            "--stop-with-process=\(ProcessInfo.processInfo.processIdentifier)",
+            "--dir=\(destination.parent.string)",
+            "--out=\(destination.basename())",
+            url.absoluteString,
+        ]
+        let stdOutPipe = Pipe()
+        process.standardOutput = stdOutPipe
+        let stdErrPipe = Pipe()
+        process.standardError = stdErrPipe
+        
+        var progress = Progress(totalUnitCount: 100)
+
+        let token = NotificationCenter.default.addObserver(
+            forName: .NSFileHandleDataAvailable, 
+            object: nil, 
+            queue: OperationQueue.main
+        ) { note in
+            let handle = note.object as! FileHandle
```

Comment:
I assume this `as!` is safe and doesn't need to be unwrapped?

Reply:
Yeah that's the case in the docs for `Notification.Name.NSFileHandleDataAvailable`, but I'll change it to return early if for some reason it isn't.